internal/receiver/discoveryreceiver/correlation.go

```diff
@@ -50,6 +50,7 @@ type correlationStore interface {
 	GetOrCreate(endpointID observer.EndpointID, receiverID component.ID) correlation
 	Attrs(endpointID observer.EndpointID) map[string]string
 	UpdateAttrs(endpointID observer.EndpointID, attrs map[string]string)
+	EmitCh() chan correlation
```

Comment:
Is it useful to have a public getter method for the channel here? Do we foresee a case where we can't just directly use the channel?

Reply:
I'm just following the pattern of how this interface is used. If I want to use it directly, I'd need to convert itto the underlying struct. I don't like having interfaces with one implementation used in the same package. I would like to remove the interface altogether, but it's out of the scope